### PR TITLE
Capture a few notes from our working conventions

### PR DESCRIPTION
**Pull request by goldfirere**
Merging this will allow me to stub out the section in our internal page about working conventions.

### CODE REVIEW

**Comment by goldfirere** [comment]
Review: @ccasin

File: HACKING.jst.md [hunk 1]
```diff
@@ -93,6 +93,15 @@ exact structural subterms. If you, for example, print an attribute specially
...
 time.
 
+Before building
+---------------
+
+You will need to install several libraries. This command may work:
+
+```
+opam install fix ocp-indent bechamel-js alcotest campl-streams fpath either dune-build-info uuseg ocaml-version
+```
```
Comment (ccasin) [inline HACKING.jst.md:103]: and `menhir.20210419`, probably?

File: HACKING.jst.md [hunk 2]
```diff
@@ -139,7 +148,10 @@ Testing
 -------
 
 To just run your built ocamlformat against a file manually, run
-`dune exec ocamlformat -- path/to/your/file`.
+`dune exec ocamlformat -- --enable-outside-detected-project path/to/your/file.ml`.
+
+If you want to see the parsed source ocamlformat produces internally, use
+`dune exec -- tools/printast/printasl.exe path/to/your/file.ml`.
```
Comment (ccasin) [inline HACKING.jst.md]:
```suggestion
`dune exec -- tools/printast/printast.exe path/to/your/file.ml`.
```

**Review by ccasin — APPROVED** [review]
Looks fine, a couple minor suggestions.

**Pull request merged**